Composer and packagist support

snufkin opened the pull request:
For various modern PHP frameworks composer support is essential for proper library and dependency management. I'd like to propose that the library adopts this based on this PR.

Changes contained:
- Restructuring the repository based on the payment library provided by Stripe as a reference
- Adding a composer file with the proper requirements (TBD - e.g. PHPUnit 6 and PHP 5.6 are incompatible, it's not clear if we need to restrict the PHP version), license
- Updated the changelog to reflect the composer related updates

Once merged I'd remove the packagist entry for my fork, it was created for testing mainly anyway.

KristofMorva commented:
@snufkin why does it require PHPUnit 6 exactly? I don't see a single test anywhere.

snufkin commented:
Good point, I've removed the phpunit dependency, and also changed the name attribute. I would like to know what those "many other issues" are so I can follow up on them.

KristofMorva commented:
@snufkin I have created some PR directly in your fork, but some ideas:
- The requirement is `ext-curl` instead of `php-mod/curl`.
- The `autoload` is not working, since it's incorrect. The library itself does not contain namespaces, so `Barion` namespace does not exist, and PSR-4 cannot be done. Either add the namespaces (that would be the professional way, but I don't know if the Barion guys have time to review and test a change like that (ping @barion-vinczei @attilabotz @emery303). If not, you can use `"classmap": [ "library" ]`, which will load every file.
- The `library` `type` is the default, so you can omit that.
- About the license are you sure it's Apache, did you talk to the Barion guys? If you are not sure, just omit it, and let them fill it out if they want to.
- I don't see the goal the `authors` with this value.
- The `api` `keyword` is a little too general, I don't think it's useful.

snufkin commented:
Done or resolved:
* PRs merged and/or updated relating to CURL and PHP version restrictions. Thanks for contributing the suggested changes.
* Apache 2.0 seems to be the official licence - as it is stated in numerous files
* Keyword may be general, however it worked for Stripe, so I figured it would work for Barion.
* Authors: suggestions are welcome, I have no issues against the current statement.

ToDo:
* Fixing the autoload, based on team feedback

KristofMorva commented:
@snufkin great, thanks! :) Hopefully they'll have the time to check this out. In my opinion Composer & autoload is a must-have.
